### ISSUE TITLE
Battlemage's Cry Spell -> Attack Damage Convert not Working

### ISSUE THREAD

**Issue by Yagaton** [PathOfBuildingCommunity/PathOfBuilding#3071]
### Check version

- [X] I'm running the latest version of Path of Building and I've verified this by checking the [changelog](https://github.com/PathOfBuildingCommunity/PathOfBuilding/blob/master/CHANGELOG.md)

### Check for duplicates

- [X] I've checked for duplicate issues by using the search function of the [issue tracker](https://github.com/PathOfBuildingCommunity/PathOfBuilding/issues)

### What is the expected behaviour/value?

I chose 100 Warcry Power
Added Battlemage's Cry and Heavy Strike to my skills
When I add +10% Spell Damage on tree my Heavy Strike damage should be going up

### What is the actual behaviour/value?

DPS doesn't increase at all untill I add: 
"Increases and Reductions to Spell Damage also apply to Attacks at 150% of their value" to any item. ( which is what it should be doing at 100 Warcry power )

### How to reproduce the issue

1. Choose 100 Warcry Power
2. Add Battlemage's Cry to Skills
3. Add any Melee Skill to Skills
4. Choose any melee weapon
5. Add Spell Damage from any Source

### Build code

```shell
https://pastebin.com/iK3wgDxM
```

**Comment by Bryserker** [comment]
I encountered the same issue today. I would like to add:

It appears multiple sources of "Increases and Reductions to Spell Damage also apply to Attacks at 150% of their value" do not stack.

Adding "Increases and Reductions to Spell Damage also apply to Attacks at 150% of their value" to a random item disables the alternate quality effect of Phantasmal Earthquake.

**Comment by Yagaton** [comment]
> I encountered the same issue today. I would like to add:
> 
> It appears multiple sources of "Increases and Reductions to Spell Damage also apply to Attacks at 150% of their value" do not stack.
> 
> Adding "Increases and Reductions to Spell Damage also apply to Attacks at 150% of their value" to a random item disables the alternate quality effect of Phantasmal Earthquake.

It shouldn't stack, so that's good :)

**Comment by fremomo** [comment]
The latest introduction of Iron Will to the skill tree raises the importance of this issue. In addition, Battle Mage Cry (BMC) now completely disables the Strength bonus to all damage types, critically breaking all damage calculations with BMC equipped.

A minimal working example to reproduce the bug with the latest version (2.10.0) is given here: https://pastebin.com/ARW7eMkN

This character has 1000 Strength which translates to 200% increased melee physical damage. To simplify the numerical simulation, I allocated Iron Will to disable critical strikes and accuracy checks. To see whether Iron Will + Battle Mage Cry correctly affect all damage types, I added 50% conversion to cold.

Following a [comment made by Mark_GGG](https://www.reddit.com/r/pathofexile/comments/41kmdu/comment/cz3gf14/?utm_source=share&utm_medium=web2x&context=3) we should expect the following behaviour:

1. We have Battle Mage Cry equipped with 100 Power, i.e. we expect spell damage to apply to attacks
2. With Iron Will allocated, we expect our strength bonus to also apply to spell damage
3. Using both 1. and 2., the strength bonus should now apply to all sources of attack damage, spell damage or melee physical damage

Therefore, we expect the strength bonus to apply to all damage types in the calculations tab. However, enabling Battle Mage Cry actually reduces the strength bonus to 0 if Power >= 5. What we should see instead is increased damage for all damage types in the calculation tab (I am unsure whether its 200% or 300% increased though).

Finally, as previously mentioned in this issue, adding additional sources of spell damage does not impact the damage calculation at all.
